tests/test.sh

```diff
@@ -0,0 +1,172 @@
+#!/bin/sh
+
+set -x
+DEVFILES_DIR="$(pwd)/stacks/"
+FAILED_TESTS=""
+
+getURLs() {
+    urls=$($ODO_PATH url list | awk '{ print $3 }' | tail -n +3 | tr '\n' ' ')
+    echo "$urls"
+}
+
+# periodicaly check url till it returns expected HTTP status
+# exit after 10 tries
+waitForHTTPStatus() {
+   url=$1
+   statusCode=$2
+
+    for i in $(seq 1 10); do
+        echo "try: $i"
+        content=$(curl -i "$url")
+        echo "Checking if $url is returning HTTP $statusCode"
+        echo "$content" | grep -q -E "HTTP/[0-9.]+ $statusCode"
+        retVal=$?
+        if [ $retVal -ne 0 ]; then
+            echo "ERROR not HTTP $statusCode"
+            echo "$content"
+        else
+            echo "OK HTTP $statusCode"
+            return 0
+        fi
+        sleep 10
+    done
+    return 1
+}
+
+# run test on devfile
+# parameters:
+# - name of a component and project 
+# - path to devfile.yaml
+test() {
+    devfileName=$1
+    devfilePath=$2
+
+    # remember if there was en error
+    error=false
+
+    tmpDir=$(mktemp -d)
+    cd "$tmpDir" || return 1
+
+    $ODO_PATH project create "$devfileName" || error=true
+    if $error; then
+        echo "ERROR project create failed"
+        FAILED_TESTS="$FAILED_TESTS $devfileName"
+        return 1
+    fi
+    
+    # Get the starter project name
+    starterProject=$($YQ_PATH e '.starterProjects[0].name' $devfilePath)
+    if [ "$REGISTRY" = "local" ]; then
+      $ODO_PATH create "$devfileName" --devfile "$devfilePath" --starter $starterProject || error=true
+    else
+      $ODO_PATH create "$devfileName" --starter $starterProject || error=true
+    fi
+
+    if $error; then
+        echo "ERROR create failed"
```

Comment:
Should this be `ERROR starter project create failed`?

Reply:
I think it should be `odo create` specifically. While, we do create the starter project, what caused odo create to fail, may not be related to that. So I'll keep in generic and say "ERROR odo create failed"

Reply:
Updated.